config/routes.rb

```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
+  root 'pages#index'
+  get 'product/new'
+  get 'product/show'
+  get 'product/edit'
+  get 'product/delete'
```

Comment:
do you need these get endpoints here? you have `resources :products` below which will define the endpoints for you. 

More details here:

<img width="704" alt="Screenshot 2021-04-28 at 17 44 33" src="https://user-images.githubusercontent.com/70625532/116423526-6f668200-a849-11eb-8dfc-42b347ac95ef.png">